terraform/stacks/data_archive/byob_ica_v2.tf

```diff
@@ -1,41 +1,45 @@
 ################################################################################
-# Variables
-
-variable "analysis_data_bucket" {
-  description = "This bucket is holding production data."
-  default     = "org.umccr.data.analysis-data"
-}
-variable "analysis_data_prefix" {
-  description = "The prefix under which all analysis data is collected."
-  default     = "analysis_data/"
+# Local constants
+
+locals {
+  # The bucket holding all "production" data
+  production_data_bucket = "org.umccr.data.production"
+  # prefix used for the ICAv2 BYOB config (mapped to and accessible from ICA)
+  ica_byob_prefix = "icav2/byob/"
+  # prefix for data in the ICA "production" project
+  production_data_prefix = "${local.ica_byob_prefix}production/"
+  # prefix for analysis data in the ICA "production" project, subject to lifecycle management
+  production_temp_data_prefix = "${local.production_data_prefix}temp/"
```

Comment:
I'm not sure we'll need a temp folder in the same way we need one for v1. Unless this is v1 we are referring to here?

Reply:
There won't be temporary workflow output? 

Note: this new configuration means ALL production data would be in this bucket, also from pipelines run outside of ICA. It's basically an active "work" space for all data that's "currently" needed. All data is then supposed to be either deleted or moved to an archive.

Reply:
Did you have a play with the cross project volume/data mapping options? 
It seems possible, but not sure it's good enough for production work, i.e. I am not sure I want to rely on it as key part of the production setup.

Reply:
> Did you have a play with the cross project volume/data mapping options?

Not sure what you mean by this, 
As in just creating files in one project bundle and sharing it to another project?

Reply:
> There won't be temporary workflow output?

Nope, either we get the outputs (for a successful workflow) or we don't. All temp data is on the local filesystem and then the out/ directory is uploaded after completion of the workflow back to S3

Reply:
>As in just creating files in one project bundle and sharing it to another project?

Well, the alternative to using a single bucket (which is linked to the `production` project) is to make other buckets/volumes available to the `production` project. 
Say a workflow reads oncoanalyser data, runs some analysis/reporting and writes its result back out. If all workflows (ICA and other) use the same bucket, there is no issue with linking/access. If oncoanalyser uses a different bucket, the data would not automatically be available in the `production` project's linked storage (various options for making it available... ). Things get even more complex if we wanted to keep "research" separate from "clinical" data. 

Probably manageable, but we'd probably end up with a combination of settings to make data available (pre-signed URLs, project/data linking, shared/multiple project access, ...) and I wonder if that's worth it. I have not clear feeling for all the issues/limitations we'd likely run into.

The simpler alternative is to have ALL "active" data in the same bucket and manage it there. A bit more risk as everyone/thing will need to have access, but at least that's mainly on our end/in our control. 

So the question here is: one bucket to rule them all or divide and conquer?

Reply:
>Nope, either we get the outputs (for a successful workflow) or we don't.

I am happy to remove it if there are no use cases, but we have this in the current ICA setup/volume. 
I guess it does not hurt to have a location ready just in case?

Reply:
> I guess it does not hurt to have a location ready just in case?

Sure no harm in keeping it.  

> The simpler alternative is to have ALL "active" data in the same bucket and manage it there

I think this is the right approach. We can have validation and reference data in a different bucket linked to a different ICAv2 project.  

> A bit more risk as everyone/thing will need to have access, but at least that's mainly on our end/in our control.

It is a bit more risk. But we could lock things down almost entirely on the ICA side -- and use it almost exclusively as a workflow execution service rather than a data access point.  

Also reducing time between data creation and archive date would reduce the risk in the worst case scenario that the entire bucket is compromised. In ICAv1, we essentially have all data in the one key prefix (and it's all active), so our own single BYOB is still better than the current setup.

Reply:
Agreed!